src/bcccp/tickets/season/SeasonTicketDAO.java

```diff
@@ -3,64 +3,108 @@
 import bcccp.tickets.season.ISeasonTicket;
 import bcccp.tickets.season.IUsageRecordFactory;
 
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Date;
+
 public class SeasonTicketDAO implements ISeasonTicketDAO {
 
+	private List<ISeasonTicket> seasonTickets;
 	private IUsageRecordFactory factory;
 
 	
 	
 	public SeasonTicketDAO(IUsageRecordFactory factory) {
-		//TOD Implement constructor
+
+		this.factory = factory;
+
+		seasonTickets = new ArrayList<>();
+
 	}
 
 
 
 	@Override
 	public void registerTicket(ISeasonTicket ticket) {
-		// TODO Auto-generated method stub
+
+		seasonTickets.add(ticket);
 		
 	}
 
 
-
 	@Override
 	public void deregisterTicket(ISeasonTicket ticket) {
-		// TODO Auto-generated method stub
+
+		// 'deregister' means: remove ticket from the seasonTickets ArrayList
+
+		Iterator<ISeasonTicket> sTicketRecs = seasonTickets.iterator();
+
+		while (sTicketRecs.hasNext()) {
+
+			if (sTicketRecs.next().getId().equals(ticket.getId())) {
+
+				sTicketRecs.remove();
+			}
+		}
 		
 	}
 
 
-
 	@Override
 	public int getNumberOfTickets() {
-		// TODO Auto-generated method stub
-		return 0;
-	}
 
+		return seasonTickets.size();
+	}
 
 
 	@Override
 	public ISeasonTicket findTicketById(String ticketId) {
-		// TODO Auto-generated method stub
-		return null;
-	}
 
+		Iterator<ISeasonTicket> sTicketRecs = seasonTickets.iterator();
+
+		ISeasonTicket sTicket = null;
+
+		while (sTicketRecs.hasNext()) {
+
+			if (sTicketRecs.next().getId().equals(ticketId)) {
+
+				sTicket = sTicketRecs.next();
+			}
+			else {
+				sTicket = null;
+			}
+		}
+		return sTicket;
+	}
```

Comment:
break: inside while.
else {
 +				sTicket = null;
 +			}
this clause is redundant  ^^^ cos it was null default.